docs/CIP/CIP-XXXX/README.md

```diff
@@ -0,0 +1,272 @@
+
+### Synchronization
+
+We refer to synchronization as the process that a wallet undertakes to catch up to the current chain's tip in order to derive its latest state and proceed to transact. The afore-described `point` represents the main anchor for wallets, to be able to:
+
+- resume the sync process since their last known point
+- keep track of their current position in the blockchain
+- track and handle chain reorganizations (rollbacks)
+
+The following sequence diagram shows, how a client connects providing an array of `point`s and extended public key. The client provides more than those two bits which is described in the [`subscribe`](./messages/client/subscribe.md) message.
+Subsequently, the server tries to find an intersection given the list of `point`s.
+
+For an identified intersection, the server computes and publishes an ordered stream of events.
+This event stream provides a chronological record of on-chain events and affectively state changes for the wallet to derive its latest state.
+
+> [!NOTE]
+> The server sources client-relevant data through credential-dependent, point-specific database queries, though the implementation details are beyond this CIP's scope.
+
+![Sequence Diagram Synchronization](./images/02-Synchronization-Sequence.svg)
+
+#### Resumability
+
+Clients can resume synchronization from their last known `point`s after disconnecting. To minimize reconnection attempts, clients should provide a list of `point`s based on their most recent state, including some that may have been affected by rollbacks. This approach reduces the likelihood of the server failing to find a valid intersection, which would otherwise result in dropped connections and additional round trips.
+
+> [!NOTE]
+> More details are described in the [`subscribe`](./messages/client/subscribe.md) message.
+
+#### Rollback Handling
+
+In a [CAP](https://en.wikipedia.org/wiki/CAP_theorem) system like Cardano, which balances global consistency with availability, there is a challenge in dealing with rollback events that impact transaction finality. Regardless of whether one uses their own full node or a server provider for transaction submission - if such events are not properly handled with the help of monitoring the rolling window of [k blocks](https://plutus-apps.readthedocs.io/en/latest/plutus/explanations/rollback.html) and respective resubmission, there is no guarantee that the transaction is finalized, eg. becomes part of the immutable pefix of the Cardano blockchain.
+
+Since our wallet-optimized protocol defines that server-sent messages include the `point` message partial, the client can check if its last `point` is greater than any subsequent message `point` from the server, without the need to introduce a `rollback` specific message partial. This assumes that the client keeps a local copy of the mutuable suffix of the blockchain defined by the security parameter.
```

Comment:
```suggestion
Since our wallet-optimized protocol defines that server-sent messages include the `point` message partial, the client can check if its last `point` is greater than any subsequent message `point` from the server, without the need to introduce a `rollback` specific message partial. This assumes that the client keeps a local copy of the volatile segment of the blockchain defined by the security parameter.
```